src/Registry.php

```diff
@@ -156,21 +178,13 @@ public function get_class($type)
         if (!empty($this->classes[$type])) {
             return $this->classes[$type];
         }
+
         if (!empty($this->default[$type])) {
             return $this->default[$type];
         }
 
-        if (!isset($this->legacyClassNames[$type])) {
-            return null;
-        }
-
-        // Try again with the legacy type resolved.
-        $type = $this->legacyClassNames[$type];
-        if (!empty($this->classes[$type])) {
-            return $this->classes[$type];
-        }
-        if (!empty($this->default[$type])) {
-            return $this->default[$type];
+        if (!empty($this->legacy[$type])) {
```

Comment:
This will no longer return properly registered class:

```php
class CustomSniffer extends SimplePie\Content\Type\Sniffer {}

$registry->register(SimplePie\Content\Type\Sniffer::class, CustomSniffer::class);
var_dump($registry->get_class(`Content_Type_Sniffer`) instanceof CustomSniffer); // returns false
```